src/App.ts

```diff
@@ -6,84 +6,158 @@
+} from "italia-pagopa-api/dist/wsdl-lib/FespCdService/FespCdPortType";
 import { toExpressHandler } from "italia-ts-commons/lib/express";
-import { Configuration } from "./Configuration";
+import * as morgan from "morgan";
+import * as redis from "redis";
+import * as soap from "soap";
+import { specs as publicApiV1Specs } from "./api/public_api_pagopa";
+import { CONFIG, Configuration } from "./Configuration";
+import { GetOpenapi } from "./controllers/openapi";
 import * as PaymentController from "./controllers/restful/PaymentController";
 import { logger } from "./utils/Logger";
 
-// Define and start a WS SOAP\Restful Server
+/**
+ * Define and start an express Server
+ * to expose RESTful and SOAP endpoints for BackendApp and Proxy requests.
+ * @param {Configuration} config - The server configuration to use
+ * @return {Promise<http.Server>} The express server defined and started
+ */
 export async function startApp(config: Configuration): Promise<http.Server> {
-  logger.info("Starting Proxy PagoPa Server...");
+  logger.info("Starting Proxy PagoPA Server...");
 
-  // Define SOAP Clients necessary to send messages to PagoPa SOAP WS
-  const verificaRPTPagoPaClient = new pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient(
-    await pagoPaSoapClient.createPagamentiTelematiciPspNodoClient({
+  // Define SOAP Clients for PagoPA SOAP WS
+  // It is necessary to forward BackendApp requests to PagoPA
+  const pagoPAClient = new pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient(
+    await pagoPASoapClient.createPagamentiTelematiciPspNodoClient({
       endpoint: `${config.PAGOPA.HOST}:${config.PAGOPA.PORT}${
-        config.PAGOPA.SERVICES.PAYMENTS_CHECK
-      }`,
-      envelopeKey: PPTPortTypes.envelopeKey
+        config.PAGOPA.WS_SERVICES.PAGAMENTI
+      }?wsdl`
     })
   );
-  const attivaRPTPagoPaClient = new pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient(
-    await pagoPaSoapClient.createPagamentiTelematiciPspNodoClient({
-      endpoint: `${config.PAGOPA.HOST}:${config.PAGOPA.PORT}${
-        config.PAGOPA.SERVICES.PAYMENTS_ACTIVATION
-      }`,
-      envelopeKey: PPTPortTypes.envelopeKey
-    })
+
+  // Define a redis client necessary to handle persistent data
+  // for async payment activation process
+  const redisClient = redis.createClient(
+    config.REDIS_DB.PORT,
+    config.REDIS_DB.HOST
   );
 
-  // Define endpoints and start SOAP + RESTFUL WS
+  // Define RESTful and SOAP endpoints
   const app = express();
   app.set("port", config.CONTROLLER.PORT);
-  app.use(bodyParser.json());
-  app.use(bodyParser.urlencoded({ extended: false }));
-  setRestfulRoutes(app, config, verificaRPTPagoPaClient, attivaRPTPagoPaClient);
+  const loggerFormat =
+    ":date[iso] [info]: :method :url :status - :response-time ms";
+  app.use(morgan(loggerFormat));
+  setRestfulRoutes(app, config, redisClient, pagoPAClient);
+  getSoapServer(
+    redisClient,
+    config.PAYMENT_ACTIVATION_STATUS_TIMEOUT_SECONDS as number
```

Comment:
you have a nice `Configuration` io-ts type that will verify that `config` has the correct structure and provide you with `PAYMENT_ACTIVATION_STATUS_TIMEOUT_SECONDS` with the right type, avoiding this unsafe cast